File: labs-training-master/java/solid/solid-java-1/src/main/java/es/emergya/labs/training/solid_java_1/srp/excercise_00/CarRepository.java
package es.emergya.labs.training.solid_java_1.srp.excercise_00;


import java.util.Arrays;
import java.util.List;

public class CarRepository {

	private List<Car> databases;


	public CarRepository(){
		this.databases = Arrays
		.asList(new Car("1", "Golf III", "Volkswagen"), 
				new Car("2", "Multipla", "Fiat"),
				new Car("3", "Megane", "Renault"));

	}

    public List<Car> getCars(){
        return this.databases;
    }

	public Car getCarById(final String id) {
		
		for (Car car : databases) {
			
			if (car.getId().equals(id)) {
				return car;
			}
		}
		
		return null;
	}
}

File: labs-training-master/SolidPythonResuelto/main/srp/CarRepository.py
from Car import Car
class CarRepository():
    def __init__(self) -> None:
        self.databases:'list[Car]' = [Car("1", "Golf III", "Volkswagen"),Car("2", "Multipla", "Fiat"),Car("3", "Megane", "Renault")]
    def getCars(self):
        return self.databases
    def getCarById(self,id:str):
        for car in self.databases :
            if (car.getId()==id) :
                return  car
        return  None